### ISSUE TITLE
AsParallel<T> does not respect custom retry strategies

### ISSUE THREAD

**Issue by roryprimrose** [jonwagner/Insight.Database#242]
Creating a ReliableConnection with a custom RetryStrategy and AsParallel for dynamic DAL support does not end up using the custom RetryStrategy.

```
        public class NullStrategy : IRetryStrategy
        {
            public TResult ExecuteWithRetry<TResult>(IDbCommand commandContext, Func<TResult> func)
            {
                throw new NotImplementedException();
            }

            public Task<TResult> ExecuteWithRetryAsync<TResult>(IDbCommand commandContext, Func<Task<TResult>> func)
            {
                throw new NotImplementedException();
            }
        }

        [TestMethod]
        public void CustomStrategyTest()
        {
            var connection = Config.DatabaseConnectionSettings.Connection();
            var retryStrategy = new NullStrategy();
            var reliableConnection = new ReliableConnection(connection, retryStrategy);

            var target = reliableConnection.AsParallel<IAggregatePublisher>();

            var capacity = 200;
            var tasks = new List<Task>(capacity);

            for (var index = 0; index < capacity; index++)
            {
                var items = BuildAggregates(capacity);

                tasks.Add(target.PublishRawAggregatesAsync(items, CancellationToken.None));
            }

            try
            {
                Task.WhenAll(tasks).Wait();
            }
            catch (Exception ex)
            {
                var outcome = ((AggregateException) ex).Flatten().InnerException;

                Trace.WriteLine(outcome);

                throw;
            }
        }
```

This has been tested with the following alternatives which do correctly use the custom retry strategy.

```
            // Use .As<T> instead of .AsParallel<T>
            var target = reliableConnection.As<IAggregatePublisher>();
```

```
            // Use .ExecuteSql
            reliableConnection.ExecuteSql("blah");
```

**Comment by roryprimrose** [comment]
Browsing through the code, I suspect the issue is when the provider clones the connection, it is only taking connection string information and creating a new connection from it at which point ReliableConnection is no longer used.

**Comment by jonwagner** [comment]
This is fixed in 5.2.9
